tests/integration/julia/sandbox-mj.html

```diff
@@ -519,7 +531,7 @@
             <span data-f-bind="salesSq[4]"></span>.
         </td>
         <td class="pass">
-            pass (0.9.0). note that this only works for span, not for input (i'm assuming that's by design).
+            pass (0.10.0). note that this only works for span, not for input (i'm assuming that's by design).
```

Comment:
Ah I see. This depends on the converter -- all converters can be made
"two-way" (it'll transform response from the api -> ui, and also transform
anything you type back into some format for the api) or "one-way" (it'll
just work for displays but not the other way round).

Converters have the option of implementing two methods: "parse" and
"convert"; they can implement either one or both. If they implement both
they're two-way, if just 'convert' they're one-way. Take a look at the
'number format' converter, that's a two-way converter. You can write your
own as:

Flow.dom.converters.register({
  alias: 'twos',
  parse: function(val) { return val * 2},
  convert: function(val) { return val / 2},
});

The usual shorthand of just doing converters.register('twos', function(){
return val / 2 }); is just shorthand. It probably won't make much sense for
your 'first' converter though, since you won't have the other values in the
array to send back.

 Let me know if this makes sense.

On Wed, Jan 13, 2016 at 2:13 AM, mmrj <notifications@github.com> wrote:

> In tests/integration/julia/sandbox-mj.html
> <https://github.com/forio/flow.js/pull/74#discussion_r49510200>:
>
> > @@ -519,7 +531,7 @@
> >              <span data-f-bind="salesSq[4]"></span>.
> >          </td>
> >          <td class="pass">
> > -            pass (0.9.0). note that this only works for span, not for input (i'm assuming that's by design).
> > +            pass (0.10.0). note that this only works for span, not for input (i'm assuming that's by design).
>
> I get errors when I try to use converters with inputs, e.g. something like <input
> data-f-bind="salesSq | first"></input> to update the last element of the
> salesSq array.
>
> I'm assuming that's on purpose, but if not let me know and I can log an
> enhancement to make converters work with inputs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/forio/flow.js/pull/74/files#r49510200>.
>